### ISSUE TITLE
Documentation Does Not Match Current Azure Portal UI

### ISSUE THREAD

**Issue by markolbert** [Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2#25]
The instructions provided in the landing page do not match the current UI for the Azure Portal. For example, step 5: 

> In the list of pages for the app, select Authentication.
> In the Redirect URIs, add a redirect URL of type Web and valued https://localhost:44321/signin-oidc
> In the Advanced settings section set Logout URL to https://localhost:44321/signout-oidc
> In the Advanced settings | Implicit grant section, check ID tokens as this sample requires the Implicit grant flow to be enabled to sign-in the user.

There is a Redirect URI section, I think, but it's called Reply URLs, and the interface does not match the description there, either.

This ongoing disconnect between the state of the released Azure Portal and virtually all of GitHub and Microsoft documentation is both annoying and disturbing. Annoying because it makes the learning curve much steeper than it should be, and disturbing because it smacks of an approach to software development that has abandoned even a reasonable degree of quality control. Frankly, the level of disconnect in evidence is far higher than almost any other open source code base I deal with...and I had sort of hoped that Microsoft, and Microsoft-led projects, would demonstrate a commitment in these areas **higher** than the norm, not **below** it.

**Comment by jmprieur** [comment]
@markolbert 
I'm sorry for your frustration with our samples. I can assure you that we spend a lot of time updating them.
That said, I'm confused as for me the description above maps to the experience in the portal.

![image](https://user-images.githubusercontent.com/13203188/49533930-901c8e80-f8c0-11e8-9a7a-877b3a803973.png)

Then given that a URL is a kind of URI, I had not thought that "adding a redirect URL" would be wrong (especially the text below the **Redirect URIs** heading explains that reply URLs and redirect URLs are the same thing.).

Are you seing something different? Or otherwise what am I missing?

**Comment by markolbert** [comment]
Yep, I'm seeing something totally different. Here's what I thought was the top-level view:

![untitled-1](https://user-images.githubusercontent.com/9095610/49534318-0f38a500-f876-11e8-9ba4-b0c58dda0d70.png)

This is inside my tenant directory.

When I click on the newly-created/registered app, here's what shows up:

![untitled-2](https://user-images.githubusercontent.com/9095610/49534478-79514a00-f876-11e8-9d5d-1ae0562e8732.png)

How did you navigate to what you're seeing? Did I create/register an app in another part of the Azure Portal unrelated to what the documentation assumes?

**Comment by jmprieur** [comment]
@markolbert 
I've 2 questions for you:
- In which Azure region are you?
- When you click on **Azure Active Directory** in the vertical menu bar on the left-hand side, don't you see **App Registration (Preview)** ?

![image](https://user-images.githubusercontent.com/13203188/49534954-336ea300-f8c3-11e8-86d3-ad5951144574.png)
 
cc: @lnalepa FYI.

**Comment by lnalepa** [comment]
@markolbert Unfortunately, the preview experience is currently not available in B2C tenants. Are you able to use another tenant? I have answered your [StackOverflow question](https://stackoverflow.com/questions/53638108/confusing-documentation-for-azure-ad-v2-sample/53638478#53638478).

**Comment by markolbert** [comment]
Nope. I don't see a preview, just App Registrations (I assumed it had been released, and was no longer in preview):

![untitled-3](https://user-images.githubusercontent.com/9095610/49535537-48bedf80-f879-11e8-9601-0988d9a82366.png)

I noticed that @lnalepa has just added to this thread that the preview experience is not available in B2C tenants. Any idea on when it might be available? Or if there's a workaround for it not being available yet?

FWIW, it'd be worth putting a noticeable disclaimer in the sample docs that they don't currently work with B2C tenants. That's probably obvious to people who know what they're doing. But for those of us who are trying to learn this stuff, it's not clear why an example which sure seems like it ought to apply, doesn't.

I will also point out that this situation is an example, IMHO, of what happens when a software environment is rapidly evolving (and here I'm not talking about the Azure Portal, but the broader Azure Active Directory environment). Stuff gets "disconnected" from the current state of the art. What's worse, because all the documentation -- both current and older stuff -- comes up through online searches, it becomes extremely difficult, for newcomers and/or people who don't live and breathe a particular subsystem, to separate the wheat from the chaff. 

I'm not sure how to do it, but there needs to be a much stricter curating function for the documentation. Right now all I can do is check the date stamp on when the documentation last got published, and discount anything more than X months old...which is a really stupid way of trying to substitute for a curating process.

**Comment by parakhj** [comment]
@markolbert i'm from the Azure AD B2C product team. Are you trying to register a B2C application to authenticate using Azure AD B2C? If so, you should be using the app registration experience through the Azure AD B2C menu

See instructions here:
https://docs.microsoft.com/en-us/azure/active-directory-b2c/active-directory-b2c-app-registration

Screenshot here:
![image](https://user-images.githubusercontent.com/22688529/49550234-c5fd4b00-f89f-11e8-909a-47814484ab86.png)

**Comment by markolbert** [comment]
Hi @parakhj. Here's what I'm trying to do: I want to authenticate users via social logins (currently Google & MicrosoftAccount), and then use Graph to store certain optional user information (those additional elements are optional because they define whether a user is a member of the site I'm building -- non-members get to use certain features of the site, but members get to use everything).

I explained this a bit more over on stackoverflow, at https://stackoverflow.com/questions/53603535/authentication-vs-authorization/53606885. I'm trying to follow the suggestion that was supplied there, which is why I stumbled into this sample.
